elyra/pipeline/component.py

```diff
@@ -277,12 +279,24 @@ def _log_warning(msg: str, logger: Optional[Logger] = None):
 
 
 class ComponentParser(LoggingConfigurable):  # ABC
-    _component_platform: RuntimeProcessorType = None
+    component_platform: RuntimeProcessorType = None
     _file_types: List[str] = None
+    _parser_class_map: Dict[str, str] = {
+        'APACHE_AIRFLOW': 'elyra.pipeline.airflow.component_parser_airflow:AirflowComponentParser',
+        'KUBEFLOW_PIPELINES': 'elyra.pipeline.kfp.component_parser_kfp:KfpComponentParser'
+    }
 
-    @property
-    def component_platform(self) -> RuntimeProcessorType:
-        return self._component_platform
+    @classmethod
+    def create_instance(cls, platform: RuntimeProcessorType) -> 'ComponentParser':
+        """
+        Class method that creates the appropriate instance of ComponentParser based on platform type name.
+        """
+        try:
+            module_name, class_name = cls._parser_class_map[platform.name].split(':')
+            module = import_module(module_name)
+            return getattr(module, class_name)()
+        except Exception as e:
+            raise RuntimeError(f"Could not get appropriate ComponentParser class: {e}")
```

Comment:
This is currently mimicking the way that entrypoints loads their class instances. An alternative would be something like the below (including removal of the class-level `_parser_class_map` structure).

```
@classmethod
def create_instance(cls, platform: RuntimeProcessorType) -> 'ComponentParser':
    parser_class_map = {subclass.component_platform.name: subclass for subclass in cls.__subclasses__()}
    return parser_class_map[platform]
```